wandb/sdk/internal/job_builder.py

```diff
@@ -313,7 +313,8 @@ def _build_artifact_job_source(
             "build_context": metadata.get("build_context"),
             "dockerfile": metadata.get("dockerfile"),
         }
-        name = self._make_job_name(self._logged_code_artifact["name"])
+        artifact_basename, *_ = self._logged_code_artifact["name"].split(":")
+        name = self._make_job_name(artifact_basename)
```

Comment:
this seems like an uncaught bug that's slipped by until now -- starting in `_create_job()` ([link](https://github.com/wandb/wandb/blob/48dbe87faca6ad1f9fbfe7dd15c2a22e651cf145/wandb/sdk/launch/create_job.py#L103)), the variable `name` gets overloaded a lot (sometimes the job name, sometimes the artifact name, sometimes the artifact collection name).

With these changes, `self._logged_code_artifact["name"]` acquires a proper version suffix (e.g. `":v0"`).  This ends up causing the `artifact.name` in [test_create_job_artifact](https://github.com/wandb/wandb/blob/48dbe87faca6ad1f9fbfe7dd15c2a22e651cf145/tests/system_tests/test_launch/test_job.py#L85) to have an incorrect name that ends with `*_v0:v0`